app/controllers/drug_stocks_controller.rb

```diff
@@ -0,0 +1,75 @@
+# This controller is meant to be used from the Android app _only_,
+# hence we handle authentication ourselves from params passed from the client.
+class DrugStocksController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_action :authenticate
+  before_action :find_current_facility
+  before_action :set_for_end_of_month
+  layout false
+
+  def create
+    DrugStock.transaction do
+      safe_params[:drug_stocks].map do |drug_stock|
+        DrugStock.create!(facility: current_facility,
+                          user: current_user,
+                          protocol_drug_id: drug_stock[:protocol_drug_id],
+                          received: drug_stock[:received].presence,
+                          in_stock: drug_stock[:in_stock].presence,
+                          for_end_of_month: @for_end_of_month)
+      end
+    end
```

Comment:
We can extract this into the drug stock model no? Seems like we're always creating them in bulk with the same tx semantics, and we can also test this in a single place.

Reply:
We can extract it eventually to a service object or something, but I'd like to wait until we have more usages or feel pain here....its pretty simple code and is also tightly coupled to the controller / params structure.